riscv/src/continuations/bootloader.rs

```diff
@@ -20,11 +22,16 @@ pub const N_LEAVES_LOG: usize = MEMORY_SIZE_LOG - PAGE_SIZE_BYTES_LOG;
 pub const MERKLE_TREE_DEPTH: usize = N_LEAVES_LOG + 1;
 pub const PAGE_SIZE_BYTES: usize = 1 << PAGE_SIZE_BYTES_LOG;
 pub const PAGE_NUMBER_MASK: usize = (1 << N_LEAVES_LOG) - 1;
-pub const BOOTLOADER_INPUTS_PER_PAGE: usize = WORDS_PER_PAGE + 1 + 8 + (MERKLE_TREE_DEPTH - 1) * 8;
+pub const WORDS_PER_HASH: usize = 8;
+pub const BOOTLOADER_INPUTS_PER_PAGE: usize =
+    WORDS_PER_PAGE + 1 + WORDS_PER_HASH + (MERKLE_TREE_DEPTH - 1) * WORDS_PER_HASH;
 pub const MEMORY_HASH_START_INDEX: usize = 2 * REGISTER_NAMES.len();
-pub const NUM_PAGES_INDEX: usize = MEMORY_HASH_START_INDEX + 16;
+pub const NUM_PAGES_INDEX: usize = MEMORY_HASH_START_INDEX + WORDS_PER_HASH * 2;
 pub const PAGE_INPUTS_OFFSET: usize = NUM_PAGES_INDEX + 1;
 
+// Ensure we have enough addresses for the scratch space.
+const_assert!(PAGE_SIZE_BYTES > 512);
```

Comment:
I'm surprised we need that much? Now pages have to be at least 1KB, I think it could be helpful to decrease below that, so the assertion should be less generous if it can be.

Reply:
I just copied what you said in your comment and assumed you did the math haha

Reply:
updated to 384, pretty sure that's safe